index.js

```diff
@@ -14,18 +17,64 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    if (type !== 'self') return;
+    this.setupJsPreprocessing(registry, type);
+
+    if (type === 'self') {
```

Comment:
Not sure if we need to worry about that right now but I think we should keep it in mind. I think that's the end goal for this feature.